src/article/shared/AddFigurePanelCommand.js

```diff
@@ -0,0 +1,39 @@
+import { Command } from 'substance'
+import { findParentByType } from './nodeHelpers'
+
+export default class AddFigurePanelCommand extends Command {
+  getCommandState (params, context) {
+    return {
+      disabled: this.isDisabled(params)
+    }
+  }
+
+  execute (params, context) {
+    const collection = this._getCollection(params, context)
+    const files = params.files
+    collection.addItem({})
+    let api = context.api
+    if (files.length > 0) {
+      api._insertFigurePanel(files)
```

Comment:
... see previous comment: I think we should use the collection's addItem(), but only do that once.